library/gui/screen/src/main/java/org/quiltmc/qsl/screen/api/client/ScreenKeyboardEvents.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright 2016, 2017, 2018, 2019 FabricMC
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.screen.api.client;
+
+import org.quiltmc.qsl.base.api.event.Event;
+import org.quiltmc.qsl.base.api.event.client.ClientEventAwareListener;
+
+import net.minecraft.client.gui.screen.Screen;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+
+/**
+ * Events related to use of the keyboard in a {@link Screen}.
+ *
+ * <p>All of these events work on top of a specific screen instance.
+ * Subscriptions will only last as long as the screen object itself; they disappear once the screen is refreshed, closed or replaced.
+ * Use {@link ScreenEvents#BEFORE_INIT} to register the desired events every time it is necessary.
+ *
+ * <p>Events are fired in the following order:
+ * <pre>{@code AllowX -> BeforeX -> AfterX}</pre>
+ * If the result of the Allow event is false, then Before and After are not called.
+ *
+ * @see ScreenEvents
+ */
+@Environment(EnvType.CLIENT)
+public final class ScreenKeyboardEvents {
+	/**
+	 * An event that checks if a key press should be allowed.
+	 */
+	public static final Event<AllowKeyPress> ALLOW_KEY_PRESS = Event.create(AllowKeyPress.class, callbacks -> (screen, key, scancode, modifiers) -> {
+		for (AllowKeyPress callback : callbacks) {
+			return callback.allowKeyPress(screen, key, scancode, modifiers);
+		}
+
+		return true;
```

Comment:
This event will only ever call its first callback. Use `TriState` and return when it is not `DEFAULT` instead, like so:
```suggestion
                TriState state = TriState.DEFAULT;
		for (AllowKeyPress callback : callbacks) {
			state = callback.allowKeyPress(screen, key, scancode, modifiers);
			if (state != TriState.DEFAULT) {
			    return state;
			}
		}

		return state;
```
This also applies to the other `ALLOW_*` events in `ScreenMouseEvents`.

Reply:
Fixed.